lib/tokenlist.cpp

```diff
@@ -1019,6 +1019,17 @@ static void compilePrecedence3(Token *&tok, AST_state& state)
                     continue;
                 }
             }
+
+            Token* leftToken = tok;
+            while (Token::Match(tok->next(), ":: %name%"))
+            {
+                Token* scopeToken = tok->next(); //The ::
+                scopeToken->astOperand1(leftToken);
+                scopeToken->astOperand2(scopeToken->next());
+                Token* leftToken = scopeToken;
```

Comment:
Interestingly this doesn't cause a bug because astOperand1 traverses up the tree and adds the ancestor node rather than the node given so it ends up adding the "::" node anyway. I'll fix it anyway since it's pretty confusing and that computation is wasteful.

Reply:
aha ok. not sure why I added that bahavior.